src/messages/MOSDECSubOpRead.h

```diff
@@ -20,16 +20,23 @@
 
 class MOSDECSubOpRead : public MOSDFastDispatchOp {
 private:
-  static constexpr int HEAD_VERSION = 3;
+  static constexpr int HEAD_VERSION = 4;
   static constexpr int COMPAT_VERSION = 1;
 
 public:
   spg_t pgid;
   epoch_t map_epoch = 0, min_epoch = 0;
   ECSubRead op;
+  uint64_t cost = 0;
 
+  void compute_cost(CephContext *cct, const std::set<hobject_t> &hoids) {
+    cost = 0;
+    for(const auto& hoid : hoids) {
+      cost += op.cost(cct, hoid);
```

Comment:
Why does compute_cost here need hoids passed in?  Doesn't the ECSubRead op already know what objects it is requesting?